pkg/k8s/portforward.go

```diff
@@ -180,11 +180,25 @@ func (pf *PortForward) URLFor(path string) string {
 	return fmt.Sprintf("http://127.0.0.1:%d%s", pf.localPort, path)
 }
 
-// getLocalPort binds to a free ephemeral port and returns the port number.
+// getLocalPort is used by dashboard.go to select a port for the dashboard.
+// It first checks the availability of the default port, defined in addr.
+// If that port is taken, it binds to a free ephemeral port and returns the
+// port number.
 func getLocalPort() (int, error) {
-	ln, err := net.Listen("tcp", ":0")
+	addr := "127.0.0.1:50750"
+	ln, err := net.Listen("tcp", addr)
 	if err != nil {
-		return 0, err
+		if opError, ok := err.(*net.OpError); ok {
+			if opError.Op == "listen" &&
+				opError.Net == "tcp" &&
+				opError.Addr.String() == addr &&
+				opError.Err.Error() == "bind: address already in use" {
```

Comment:
Seeing all of these hardcoded strings makes me nervous that this code would break during an upgrade, and I don't think we need this level of error checking. If the initial call to `net.Listen` fails with an error, let's assume that the failure was a result of the port being in use, and always fall back to listening on an ephemeral port. If it wasn't a result of the port being in use, I'd expect the listen call on the ephemeral port to return the same error, at which point we can return that.

It's also the case that if we do successfully bind to 50750, then we already know the port and can return it immediately, instead of needing to read it from the listener on line 206. I recommend setting the function up like this:

```
func getLocalPort() (int, error) {
	defaultAddr := fmt.Sprintf("127.0.0.1:%d", defaultPort)
	if _, err := net.Listen("tcp", defaultAddr); err == nil {
		return defaultPort, nil
	}

	ln, err := net.Listen("tcp", ":0")
	if err != nil {
		return 0, err
	}
	defer ln.Close()
	...
```

That would require you to define a `defaultPort` const at the top of this file, like this:

```
const defaultPort = 50750
```

Reply:
Sounds good, I just pushed the updated branch, which passes all unit tests.